crates/router/src/core/payment_methods/cards.rs

```
@@ -261,6 +262,310 @@ pub async fn get_or_insert_payment_method(
     }
 }
 
+pub async fn migrate_payment_method(
+    state: routes::SessionState,
+    req: api::PaymentMethodMigrate,
+) -> errors::RouterResponse<api::PaymentMethodResponse> {
+    let merchant_id = &req.merchant_id;
+    let key_store = state
+        .store
+        .get_merchant_key_store_by_merchant_id(
+            merchant_id,
+            &state.store.get_master_key().to_vec().into(),
+        )
+        .await
+        .to_not_found_response(errors::ApiErrorResponse::MerchantAccountNotFound)?;
+
+    let merchant_account = state
+        .store
+        .find_merchant_account_by_merchant_id(merchant_id, &key_store)
+        .await
+        .to_not_found_response(errors::ApiErrorResponse::MerchantAccountNotFound)?;
+
+    let card_details = req
+        .card
+        .as_ref()
+        .ok_or(errors::ApiErrorResponse::InvalidRequestData {
+            message: "Missing field card".to_owned(),
+        })
+        .attach_printable("Card details can not be null")?;
+
+    let card_number_validation_result = cards::CardNumber::from_str(&card_details.card_number);
+
+    match card_number_validation_result {
+        Ok(card_number) => {
+            let payment_method_create_request =
+                api::PaymentMethodCreate::get_payment_method_create_from_payment_method_migrate(
+                    card_number,
+                    &req,
+                );
+            get_client_secret_or_add_payment_method(
+                state,
+                payment_method_create_request,
+                &merchant_account,
+                &key_store,
+            )
+            .await
+        }
+        Err(_) => {
```

Don't discard the error, and have a error log.